tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -76,24 +79,38 @@ spec:
         build_tags=$(jq '.fbc.buildTags // []' ${DATA_FILE})
         add_arches=$(jq '.fbc.addArches // []' ${DATA_FILE})
         hotfix=$(jq -r '.fbc.hotfix // "false"' ${DATA_FILE})
+        pre_ga=$(jq -r '.fbc.preGA // "false"' ${DATA_FILE})
         staged_index=$(jq -r '.fbc.stagedIndex // "false"' ${DATA_FILE})
+        product_name=$(jq -r '.fbc.productName // ""' ${DATA_FILE})
+        product_version=$(jq -r '.fbc.productVersion // ""' ${DATA_FILE})
         build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
             '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
-        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
         fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH})
 
         timestamp_format=$(jq -r '.fbc.timestampFormat // "%s"' ${DATA_FILE})
         timestamp=$(date "+${timestamp_format}")
 
+        # default target_index
+        target_index=$(params.targetIndex)
+
+        # the target_index is modified when the pipelinerun is a for `hotfix` or a `pre-GA` release
         if [ "${hotfix}" == "true" ]; then
           issue_id=$(jq -r '.fbc.issueId // empty' ${DATA_FILE})
           if [ -z "${issue_id}" ]; then
-            echo "Hotfix releases requires the issue id."
+            echo "Hotfix releases requires the issue id set in the `fbc.issueId` key of the ReleasePlanAdmission " \
+                 "spec.data field"
             exit 1
           fi
-          tag="${issue_id}-${timestamp}"
-          target_index="${target_index}-${tag}"
+          target_index="${target_index}-${issue_id}-${timestamp}"
+        elif [ "${pre_ga}" == "true" ]; then
+          if [ -z "${product_name}" ] || [ -z "${product_version}" ]; then
```

Comment:
Nit: Those two conditions could be combined as `[ -z "${product_name}" -o -z "${product_version}" ]`

Reply:
Cool. I knew it, just (for no reason) I like more the syntax used. But I can change to `-o` instead. Thanks!

Reply:
OK, it's up to you :)